NextJs/pages/index.js

```diff
@@ -1,87 +1,168 @@
-                const callbacks = {
-                    onSuccess: (res) => {
-                        console.log('onSuccess', JSON.parse(res))
-                    },
-                    onAbort: () => console.log('onAbort callback'),
-                    onError: (err) => console.log(JSON.stringify(err))
-                };
+        <Dropdown.Menu>
+          {menu.map((item, index) => {
+            return (
+              <Dropdown.Item key={index} eventKey={index}>{`${item.name} - $${
+                item.price / 100
+              }`}</Dropdown.Item>
+            );
+          })}
+        </Dropdown.Menu>
+      </Dropdown>
+    );
+  };
 
-                // Initializing widget with order information
-                // eslint-disable-next-line no-undef
-                HolaCashCheckout.configure(
-                    { order_id: response?.data?.order_information?.order_id },
-                    callbacks
-                );
-            }
-        } catch {
-            setOrderLoading(false);
+  // Creating an order https://developers.holacash.mx/openapi/cash/#tag/order
+  const generateOrder = async (item) => {
+    setOrderLoading(true);
+    try {
+      const response = await axios.post(
+        "https://sandbox.api.holacash.mx/v2/order",
+        {
+          order_total_amount: {
+            amount: item.price,
+            currency_code: "MXN",
+          },
+          description: item.name,
+        },
+        {
+          headers: {
+            "X-Api-Client-Key": merchantPublicKey,
+            "Content-Type": "application/json",
+          },
         }
-    };
+      );
+      console.log(response);
+      setOrderLoading(false);
 
-    return (
-        <div className="App">
-            <Head>
-                <title>Create Next App</title>
-                <meta name="description" content="Generated by create next app" />
-                <link rel="icon" href="/favicon.ico" />
-            </Head>
-            <Script src="https://widget.connect.play.holacash.mx/connect.min.js" data-public-key="pub_sandbox_96qsBHrP.IIVG6xPaGkLIneXeekspk9muIAfYaesn" id="holacash-connect" strategy="beforeInteractive" />
-            <button onClick={() => generateOrder()}>
-                Pagar
-            </button>
-            {/* Creating Button object  */}
-            <div id="instant-holacash-checkout-button">
-                {!orderLoading && orderId ? (
-                    <object
-                        id="checkout-button"
-                        data={`https://sandbox.api.holacash.mx/v2/checkout/button?public_key=${merchantPublicKey}`}
-                    />
-                ) : null}
-            </div>
-        </div>
-    );
+      // verifying correct response from create order
+      if (response?.data?.order_information?.order_id) {
+        setOrderId(response?.data?.order_information?.order_id);
+        console.log("ORDER ID:", response?.data?.order_information?.order_id);
+        const callbacks = {
```

Comment:
@ali-cash `// callbacks can be used so that you as a merchant can react however you want to certain events fired by the widget. For example, if the widget gets closed, the `onAbort()` method will be called and you can ask the user why...`

Reply:
onSuccess --> we call this callback when the charge is successfull. The parameter passed in is the response of the charge API
onAbort --> when the user closes the widget (modal/popup)
onError --> when the charge fails. The parameter is the error object